src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProvider.cs

```diff
@@ -258,6 +265,100 @@ private async Task<IReadOnlyList<SecretListEntry>> FetchAllSecretsAsync(Cancella
             }
             return configuration;
         }
+        
+        private static List<List<SecretListEntry>> ChunkList(IReadOnlyList<SecretListEntry> source,
+            Func<SecretListEntry, bool> optionsSecretFilter, int chunkSize)
+        {
+            // This is for sake of cleanliness vs getting 'fancy' with things.
+            // We can always optimize later.
+            return source
+                .Where(sle=> !optionsSecretFilter(sle))
+                .Select((item, index) => new { item, index })
+                .GroupBy(x => x.index / chunkSize)
```

Comment:
I think later versions of .net include a chunk method but not sure if it's compatible with the targeted frameworks.

Reply:
Yeah, NET 6.0 and up - https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.chunk?view=net-9.0

Reply:
Per this not being part of `netstandard2.0` decided to leave as-is. Doubly so since currently lib is only published that way for now.

In future if desired, folks can do a `#if NET6_0_OR_GREATER` dance for the more optimized call in separate PR.

Similar for multitargeting; I -thought- about it for a moment but decided better to keep changes small. (That said I may do it in separate PR if someone doesn't beat me to it)

Reply:
Ignoring the multi-targeting changes required, here it is with `#if NET6_0_OR_GREATER`:

```
        private static IEnumerable<SecretListEntry[]> ChunkList(IReadOnlyList<SecretListEntry> source,
            Func<SecretListEntry, bool> optionsSecretFilter, int chunkSize)
        {
#if NET6_0_OR_GREATER
            return source.Where(optionsSecretFilter).Chunk(20);
#else
            // This is for sake of cleanliness vs getting 'fancy' with things.
            // We can always optimize later.
            return source
                .Where(optionsSecretFilter)
                .Select(static (item, index) => (item, index))
                .GroupBy(x => x.index / chunkSize)
                .Select(static group => group.Select(static x => x.item).ToArray())
                .ToList();
#endif
        }
```